cdap-proto/src/main/java/co/cask/cdap/proto/id/NamespacedId.java

```diff
@@ -15,9 +15,16 @@
  */
 package co.cask.cdap.proto.id;
 
+import co.cask.cdap.proto.element.EntityType;
+
 /**
  * An {@link EntityId} which belongs to a namespace.
  */
-public interface NamespacedId {
-  String getNamespace();
+public abstract class NamespacedId extends EntityId {
```

Comment:
I don't like this name of the class. Probably better call it `EntityIdWithNamespace` or `NamespacedEntityId`.